src/pages/GivePage/index.tsx

```diff
@@ -184,19 +185,19 @@ const GivePage = () => {
   };
 
   // adjustGift adjusts a gift by an amount if it is allowed wrt the max give, then schedules saving
-  const adjustGift = (recipientId: number, amount: number) => {
+  const adjustGift = (recipientId: number, amount: number | null) => {
```

Comment:
Could we continue to require this to be non-null and just make sure to not fire the adjustGift event if the number is empty? Would be nice to not have to pass forward the optional logic all across this...

Reply:
We update the `gift.tokens` in the `adjustGift()` and I could not find a clean way to tell that function that the input is empty. We want to allow `gift.tokens` to be `null` or `undefined` so that we can set the `TextField` value to empty string in
```typescript
<TextField
  data-testid="tokenCount"
  value={gift.tokens ?? ''}
  // ...
/>
```
I've initially added a new param `isEmpty` to that function signature, but that felt a bit overkill, hence why I've ended up with `amount` being `number | null` precisely  -- as you've correctly mentioned -- to tell the `adjustGift` that the number is empty.

Feel free to play give it a go, a few things to test:
1. Delete all the numbers;
2. Type `0` into an empty field;
3. Type any number as long as they are less than the total gives left in the empty field.